src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void getOperation(List<FruitTransaction> transactions) {
+        List<FruitTransaction> transactionsSelected = transactions.stream()
+                .filter(a -> a.getOperation() == FruitTransaction.Operation.SUPPLY)
+                .toList();
+
+        for (FruitTransaction fruitTransaction : transactionsSelected) {
+            int newValue = Storage.fruits.get(fruitTransaction.getFruit())
+                    + fruitTransaction.getQuantity();
+            Storage.fruits.put(fruitTransaction.getFruit(), newValue);
+        }
+    }

```

Comment:
The `SupplyOperation` class should not directly access and manipulate the `Storage` class. This violates the SOLID principles, specifically the Single Responsibility Principle, because it couples the operation logic with the storage mechanism. Consider abstracting the storage access behind a service layer that this operation would then call.